docs/custom-filters.adoc

```diff
@@ -45,23 +157,173 @@ Instances are specific to the channel between a single client and a single broke
 
 It exists while the client remains connected.
 
+=== Filter Patterns
+
+Kroxylicious Protocol Filters support several patterns:
+
+1. <<Synchronously intercepting/mutating Requests and Responses>>
+2. <<Responding from a Request Filter without Proxying>>
+3. <<Sending new messages to Kafka that are not visible to the Client>>
+4. <<Filtering specific API Versions>>
+
+==== Synchronously intercepting/mutating Requests and Responses
+
+This is a common pattern, we want to inspect or modify a message:
+
+[source,java]
+----
+public class SampleFetchResponseFilter implements FetchResponseFilter {
+    @Override
+    public void onFetchResponse(short apiVersion,
+                                ResponseHeaderData header,
+                                FetchResponseData response,
+                                KrpcFilterContext context) {
+        mutateResponse(response, context);
+        context.forwardResponse(header, response);
+    }
+}
+----
+
+==== Responding from a Request Filter without Proxying
+
+In some cases we may wish to not forward a message from the client to Kafka. Instead, we want to generate a response
+message in Kroxylicious and send it back to the client. For example:
```

Comment:
You need to get across that the generated response goes back *towards* the client, through the portion of the filter chain upstream of where we are.  Your wording makes it sounds like it goes straight to the client.

Reply:
Done, have added a diagram and clarified, trying to use `toward` and `flows` instead of saying `sent to`.